ui/src/services/loki/api.js

```diff
@@ -0,0 +1,82 @@
+//@flow
+import ApiClient from '../ApiClient';
+
+let lokiApiClient: ?ApiClient = null;
+
+export function initialize(apiUrl: string) {
+  lokiApiClient = new ApiClient({ apiUrl });
+}
+
+type MetrixValue = {
+  metric: {
+    [labelName: string]: string,
+  },
+  values: [number, string],
+}[];
+type StreamValue = {
+  stream: {
+    [labelName: string]: string,
+  },
+  values: [string, string],
+}[];
+type LokiQueryResult = {
+  status: 'success',
+  data: {
+    resultType: 'matrix' | 'streams',
+    result: MetrixValue | StreamValue,
+    // Those statistics allow users to understand the amount of data processed and at which speed.
+    // https://grafana.com/docs/loki/latest/api/#Statistics
+    stats: {},
+  },
+};
+
+// by default to get the last 7day alerts from Loki
+export function getLast7DaysAlerts(): LokiQueryResult {
+  if (!lokiApiClient) {
+    throw new Error('lokiApiClient should be defined');
+  }
+
+  const query = "{container='alertmanager-logger'}";
+  const start = new Date(
+    new Date().getTime() - 7 * 24 * 60 * 60 * 1000,
+  ).toISOString();
+  const end = new Date().toISOString();
+
+  return lokiApiClient
+    .get(`/api/v1/query_range?start=${start}&end=${end}&query=${query}`)
+    .then((resolve) => {
+      if (resolve.error) {
+        throw resolve.error;
+      }
+      return resolve;
+    });
+}
+
+// customise the `start` and `end` time to retrieve the history alerts
+export function getAlertsLoki(start: string, end: string): LokiQueryResult {
```

Comment:
It should return a promise, also please ensure flow was reporting this issue as it should have reported it.
```suggestion
export function getAlertsLoki(start: string, end: string): Promise<LokiQueryResult> {
```